### PR TITLE
Add *.textproto support

### PR DESCRIPTION
**Pull request by vorburger**
This allows clang-format to format files of the format https://protobuf.dev/reference/protobuf/textformat-spec/.

It also requires a change to https://github.com/pre-commit/identify.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
     - run: pip install pre-commit-mirror-maker
     - run: git config --global user.name 'Github Actions'
     - run: git config --global user.email '41898282+github-actions[bot]@users.noreply.github.com'
-    - run: pre-commit-mirror . --language=python --package-name=clang-format --id=clang-format --entry='clang-format -i' --types-or c++ --types-or c --types-or 'c#' --types-or 'cuda' --types-or 'java' --types-or 'javascript' --types-or 'json' --types-or 'objective-c' --types-or 'proto' --args=-style=file
+    - run: pre-commit-mirror . --language=python --package-name=clang-format --id=clang-format --entry='clang-format -i' --types-or c++ --types-or c --types-or 'c#' --types-or 'cuda' --types-or 'java' --types-or 'javascript' --types-or 'json' --types-or 'objective-c' --types-or 'proto' --types-or 'proto' --args=-style=file
```
Comment (asottile) [inline .github/workflows/main.yml:18]: this is wrong

File: .pre-commit-hooks.yaml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     description: ''
     entry: clang-format -i
     language: python
-    'types_or': [c++, c, c#, cuda, java, javascript, json, objective-c, proto]
+    'types_or': [c++, c, c#, cuda, java, javascript, json, objective-c, proto, textproto]
```
Comment (asottile) [inline .pre-commit-hooks.yaml:6]: this is generated

Comment (vorburger) [reply .pre-commit-hooks.yaml]: Huh? By what? But OK, I've reverted it...

File: .github/workflows/main.yml (hunk 1 above)
Comment (vorburger) [reply .github/workflows/main.yml]: Duh, yes, I meant this to be `textproto` not `proto` - now fixed. Thanks for spotting.

**Review by asottile — APPROVED** [review]

**Pull request merged**